src/services/syncDb.ts

```diff
@@ -0,0 +1,43 @@
+import * as path from 'path';
+
+import * as fs from '../util/fs';
+import OcliffLogger from '../domain/OcliffLogger';
+import SyncDbOptions from '../domain/SyncDbOptions';
+import { CONNECTIONS_FILENAME } from '../constants';
+import { resolveConnectionsFromEnv } from '../config';
+
+/**
+ * Generates connections.sync-db.json file.
+ *
+ * @param {OcliffLogger} logger
+ * @returns {Promise<void>}
+ */
+async function generateConnection(logger: OcliffLogger): Promise<void> {
+  try {
+    const filePath = path.resolve(process.cwd(), CONNECTIONS_FILENAME);
+
+    const connections = resolveConnectionsFromEnv();
+    await fs.write(filePath, JSON.stringify({ connections })).then(() => {
```

Comment:
No need to use `await` and `then` together.
```suggestion
    const contents = JSON.stringify({ connections });
    await fs.write(filePath, contents);
    
    logger.info(`Generated file: ${CONNECTIONS_FILENAME}`);
```